docker/puppet-agent-alpine/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM alpine:3.4
+
+ARG version=5.5.1
+ARG vcs_ref
+ARG build_date
+
+ENV PUPPET_AGENT_VERSION="$version"
+ENV PUPPET_VERSION="~> 5.5.1"
+ENV FACTER_VERSION="2.5.1"
+
+LABEL org.label-schema.maintainer="Puppet Release Team <release@puppet.com>" \
+      org.label-schema.vendor="Puppet" \
+      org.label-schema.url="https://github.com/puppetlabs/puppet-agent" \
+      org.label-schema.name="Puppet Agent (Alpine)" \
+      org.label-schema.license="Apache-2.0" \
+      org.label-schema.version="$PUPPET_AGENT_VERSION" \
+      org.label-schema.vcs-url="https://github.com/puppetlabs/puppet-agent" \
+      org.label-schema.vcs-ref="$vcs_ref" \
+      org.label-schema.build-date="$build_date" \
+      org.label-schema.schema-version="1.0" \
+      org.label-schema.dockerfile="/Dockerfile"
+
+RUN apk add --no-cache \
+      ca-certificates \
+      pciutils \
+      ruby \
+      ruby-irb \
+      ruby-rdoc \
+      && \
+    echo http://dl-4.alpinelinux.org/alpine/edge/community/ >> /etc/apk/repositories && \
+    apk add --no-cache shadow && \
+    gem install puppet:"$PUPPET_VERSION" facter:"$FACTER_VERSION" && \
```

Comment:
With Puppet6, we will we need any other T&P/Modules now that they are separate? 

Also, does Puppet 6 work with Facter 2.5?

Also, why is this not Puppet 6?

Reply:
And *then* I saw the branch.

Should this whole thing just move to the 6 branch?

Reply:
Are we going to need any 5.x containers? Since 6 is out now, I could move this, mostly was targeting 5.x since we have containers for the other projects in that stream